test/controller/confirm.company.spec.unit.ts

```diff
@@ -97,13 +99,16 @@ describe("confirm company tests", () => {
     mockCreateNewObjection.mockReset();
     mockCreateNewObjection.mockImplementation(() => OBJECTION_ID);
 
+    mockGetObjectCreate.mockReset();
+    mockGetObjectCreate.mockImplementation(() =>  dummyObjectionCreate );
+
     const response = await request(app).post(OBJECTIONS_CONFIRM_COMPANY)
       .set("Referer", "/")
       .set("Cookie", [`${COOKIE_NAME}=123`]);
 
     expect(mockGetObjectionSessionValue).toHaveBeenCalledTimes(1);
     expect(mockSetObjectionSessionValue).toHaveBeenCalledWith(SESSION, SESSION_OBJECTION_ID, OBJECTION_ID);
-    expect(mockCreateNewObjection).toHaveBeenCalledWith(dummyCompanyProfile.companyNumber, undefined);
+    expect(mockCreateNewObjection).toHaveBeenCalledWith(dummyCompanyProfile.companyNumber, undefined, dummyObjectionCreate);
```

Comment:
we need to test that the delete has happened as well

Reply:
Added a line to test this.